sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```
@@ -1250,7 +1301,12 @@ private long insertDocumentHistoryIntoExistingTree(SQLDatabase db, BasicDocument
         options.data = newRevision.asBytes();
         options.available = true;
         options.copyAttachments = false;
-        long sequence = insertRevision(db,options);
+        long sequence;
+        try {
+            sequence = insertRevision(db, options);
+        } catch (AttachmentException e){
+            throw new RuntimeException("Attachments should not be copied when inserting document history",e);
```

Yeah, perhaps the wording could be improved, but if we hit that catch clause we should throw a RuntimeException rather than swallowing it